sdk/appconfiguration/Azure.Data.AppConfiguration/src/AuthenticationPolicy.cs

```diff
@@ -23,62 +23,82 @@ public AuthenticationPolicy(string credential, byte[] secret)
+            using (var contentHashStream = new CryptoStream(memoryStream, alg, CryptoStreamMode.Write))
             {
-                Uri uri = message.Request.Uri.ToUri();
-                var host = uri.Host;
-                var pathAndQuery = uri.PathAndQuery;
-
-                string method = message.Request.Method.Method;
-                DateTimeOffset utcNow = DateTimeOffset.UtcNow;
-                var utcNowString = utcNow.ToString("r", CultureInfo.InvariantCulture);
-                var stringToSign = $"{method}\n{pathAndQuery}\n{utcNowString};{host};{contentHash}";
-                var signature = Convert.ToBase64String(hmac.ComputeHash(Encoding.ASCII.GetBytes(stringToSign))); // Calculate the signature
-                string signedHeaders = "date;host;x-ms-content-sha256"; // Semicolon separated header names
-
-                message.Request.Headers.SetValue("Date", utcNowString);
-                message.Request.Headers.SetValue("x-ms-content-sha256", contentHash);
-                message.Request.Headers.SetValue("Authorization", $"HMAC-SHA256 Credential={_credential}&SignedHeaders={signedHeaders}&Signature={signature}");
+                if (message.Request.Content != null)
+                {
+                    await message.Request.Content.WriteToAsync(contentHashStream, message.CancellationToken).ConfigureAwait(false);
+                }
             }
+
+            return Convert.ToBase64String(alg.Hash);
         }
 
-        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        private void AddHeaders(HttpMessage message, string contentHash)
         {
-            ProcessAsync(message, async: false).GetAwaiter().GetResult();
+            var utcNowString = DateTimeOffset.UtcNow.ToString("r", CultureInfo.InvariantCulture);
+            var authorization = GetAuthorizationHeader(message.Request, contentHash, utcNowString);
 
-            ProcessNext(message, pipeline);
+            message.Request.Headers.SetValue("x-ms-content-sha256", contentHash);
+            message.Request.Headers.SetValue(HttpHeader.Names.Date, utcNowString);
+            message.Request.Headers.SetValue(HttpHeader.Names.Authorization, authorization);
+        }
+
+        private string GetAuthorizationHeader(Request request, string contentHash, string date) {
+            const string signedHeaders = "date;host;x-ms-content-sha256"; // Semicolon separated header names
+
+            var uri = request.Uri.ToUri();
+            var host = uri.Host;
+            var pathAndQuery = uri.PathAndQuery;
+            var method = request.Method.Method;
+
+            var stringToSign = $"{method}\n{pathAndQuery}\n{date};{host};{contentHash}";
+            var signature = ComputeHash(stringToSign); // Calculate the signature
+            return $"HMAC-SHA256 Credential={_credential}&SignedHeaders={signedHeaders}&Signature={signature}";
+        }
+
+        private string ComputeHash(string value)
```

Comment:
With so few lines in this method - why split it into a separate method?

Reply:
To reduce the scope of `using` declaration. IMO, it is easier to understand code in future when exception handlers and resource scopes are reduced to the sufficient minimum

Reply:
OK.